runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -4758,10 +4765,26 @@ J9::Z::TreeEvaluator::multianewArrayEvaluator(TR::Node * node, TR::CodeGenerator
 
    bool use64BitClasses = comp->target().is64Bit() && !TR::Compiler->om.generateCompressedObjectHeaders();
 
-   // Init class field, then jump to end of ICF
    generateRXInstruction(cg, use64BitClasses ? TR::InstOpCode::STG : TR::InstOpCode::ST, node, classReg, generateS390MemoryReference(targetReg, TR::Compiler->om.offsetOfObjectVftField(), cg));
-   cursor = generateS390BranchInstruction(cg, TR::InstOpCode::BRC, TR::InstOpCode::COND_BRC, node, cFlowRegionEnd);
-   iComment("Init class field and jump to end of ICF.");
+#if defined(TR_TARGET_64BIT)
+   if (isIndexableDataAddrPresent)
+      {
+      TR_ASSERT_FATAL_WITH_NODE(node
+         , IS_32BIT_SIGNED(fej9->getOffsetOfDiscontiguousDataAddrField() - fej9->getOffsetOfContiguousDataAddrField())
+         , "dataAddrFieldOffset is too big for the instruction.");
+
+      // Load dataAddr slot offset difference since 0 size arrays are treated as discontiguous.
+      generateRILInstruction(cg
+         , TR::InstOpCode::LGFI
+         , node
+         , temp1Reg
+         , static_cast<int32_t>(fej9->getOffsetOfDiscontiguousDataAddrField() - fej9->getOffsetOfContiguousDataAddrField()));
+      cursor = generateS390BranchInstruction(cg, TR::InstOpCode::BRC, TR::InstOpCode::COND_BRC, node, populateFirstDimDataAddrSlot);
+      }
+   else
+      cursor = generateS390BranchInstruction(cg, TR::InstOpCode::BRC, TR::InstOpCode::COND_BRC, node, cFlowRegionEnd);
```

Comment:
So in case of non 64-Bit, we would not generate this instruction, which looking at the original code was there, Is this intended? If yes, Can you put a small comment in the code or here in the PR to explain why ?

Reply:
ah sorry that is not intended, we want to generate the instruction in all cases when dataAddr pointer is not enabled. I'll correct the code.